### DISCUSSION TITLE
Getting typescript errors in generated fragment types, help?

### DISCUSSION THREAD

**Discussion by jkerrwilson** [dotansimha/graphql-code-generator#8760] [Q&A]
I'm getting the following typescript error on every fragment type generated, and I really can't find any information on what I need to change or how to fix it.

 ```
TypeScript error in <...>/src/_gql_/graphql.ts(2292,1310):
Property 'definitions' does not exist on type 'TypedDocumentNode<FragmentName, unknown>'.  TS2339
```

This is in a react app using apollo-client.

I'm currently attempting to generate TS types for an existing but currently still fairly small project, so the error currently shows up for around 50 fragments.

The error occurs on lines where fragments are spread into the parent query/mutation types, e.g.,
```ts
export const SomeQueryDocument = {"kind":"Document","definitions": <Lots of properties> ,...SomeFragmentDoc.definitions,...AnotherFragmentDoc.definitions]} as unknown as DocumentNode<SomeQuery, SomeQueryVariables>;
```

Can anyone provide insight on what is going wrong? the query/mutation/fragment setup in the code matches with those found in the relevant documetation blog post (https://the-guild.dev/blog/unleash-the-power-of-fragments-with-graphql-codegen). Only thing I have been able to find is this old issue (https://github.com/dotansimha/graphql-code-generator/issues/528) which has similar error messages but otherwise appears unrelated.

Config:
```js
const config: CodegenConfig = {
  schema: 'https://localhost:5001/graphql',
  documents: ['src/**/*.tsx'],
  ignoreNoDocuments: true,
  generates: {
    './src/_gql_/': {
      preset: 'client',
      plugins: [],
      presetConfig: {
        gqlTagName: 'gql',
        fragmentMasking: false,
      },
      config: {
        dedupeFragments: true,
      }
    }
  }
}
```
dependency versions:
```json
"@graphql-codegen/cli": "^2.16.1",
"typescript": "^4.9.4",
"@apollo/client": "^3.7.3",
```

**Comment by saihaj** [comment]
can you share a minimal reproduction?
